Source/Particles/Deposition/VarianceAccumulationBuffer.cpp

```diff
@@ -0,0 +1,212 @@
+/* Copyright 2025 The WarpX Community
+ *
+ * This file is part of WarpX.
+ *
+ * Authors: S. Eric Clark (Helion Energy)
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#include "Particles/Deposition/VarianceAccumulationBuffer.H"
+#include "Particles/Deposition/TemperatureDeposition.H"
+#include "Parallelization/WarpXSumGuardCells.H"
+#include "Fields.H"
+
+#include <ablastr/utils/Communication.H>
+
+#include <AMReX.H>
+#include <AMReX_REAL.H>
+
+using namespace amrex::literals;
+using namespace warpx::particles::deposition;
+
+VarianceAccumulationBuffer::VarianceAccumulationBuffer (ablastr::fields::MultiLevelVectorField const& T_vf, std::string const& species_name )
+{
+    using ablastr::fields::Direction;
+    auto & warpx = WarpX::GetInstance();
+
+    const int ncomps = 1;
+
+    for (int lev = 0; lev <= warpx.finestLevel(); ++lev) {
```

Comment:
It looks likes this could be simplified by using a loop over the dimensions,
```
for (int idir=0 ; idir < 3 ; idir++) {
    amrex::BoxArray const& ba = T_vf[lev][Direction{idir}]->boxArray();
    amrex::DistributionMapping const& dm = T_vf[lev][Direction{idir}]->DistributionMap();
    amrex::IntVect const& ng = T_vf[lev][Direction{idir}]->nGrowVect();
    warpx.m_fields.alloc_init("w_" + species_name, Direction{idir}, lev, ba, dm, ncomps, ng, 0.0_rt);
    ...
}
```

Reply:
I have updated the VarianceAccumulationBuffer to store the speciens name and added a getter method to retrieve the MF pointers from the buffer object that grabs a fresh reference from the MF Register.